src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.lib.Inject;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        ShoppingCart managedShoppingCart = shoppingCartDao.getByUser(shoppingCart.getUser());
+        User user = managedShoppingCart.getUser();
+        List<Ticket> tickets = managedShoppingCart.getTickets();
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
+        order.setOrderDate(LocalDateTime.now());
+        return orderDao.add(order);
+    }
```

Comment:
According to task checklist:
When passing tickets from shoppingCart to order - pass the copy of this list(you can use ArrayList constructor for this); The reason is that Hibernate will automatically delete these tickets from shoppingCart(in DB) after noticing them in our order object. Let's do this manually using clear() method from ShoppingCartService after the order with tickets has been created.